examples/image_segmentation/android/app/src/main/java/com/google/mediapipe/examples/imagesegmenter/ImageSegmenterHelper.kt

```diff
@@ -0,0 +1,245 @@
+            )
+        }
+
+        try {
+            val baseOptions = baseOptionsBuilder.build()
+            val optionsBuilder = ImageSegmenter.ImageSegmenterOptions.builder()
+                .setRunningMode(runningMode).setBaseOptions(baseOptions)
+                .setOutputType(ImageSegmenter.ImageSegmenterOptions.OutputType.CATEGORY_MASK)
+                .setResultListener(this::returnLivestreamResult)
+                .setErrorListener(this::returnLivestreamError)
+            val options = optionsBuilder.build()
+            imagesegmenter = ImageSegmenter.createFromOptions(context, options)
+        } catch (e: IllegalStateException) {
+            imageSegmenterListener?.onError(
+                "Image segmenter failed to initialize. See error logs for details"
+            )
+            Log.e(
+                TAG,
+                "Image segmenter failed to load model with error: " + e.message
+            )
+        } catch (e: RuntimeException) {
+            // This occurs if the model being used does not support GPU
+            imageSegmenterListener?.onError(
+                "Image segmenter failed to initialize. See error logs for " + "details",
+                GPU_ERROR
+            )
+            Log.e(
+                TAG,
+                "Image segmenter failed to load model with error: " + e.message
+            )
+        }
+    }
+
+    // Runs image segmentation on live streaming cameras frame-by-frame and
+    // returns the results asynchronously to the caller.
+    fun segmentsLiveStreamFrame(imageProxy: ImageProxy) {
+        if (runningMode != RunningMode.LIVE_STREAM) {
+            throw IllegalArgumentException(
+                "Attempting to call segmentsLiveStreamFrame" + " while not using RunningMode.LIVE_STREAM"
+            )
+        }
+
+        val frameTime = SystemClock.uptimeMillis()
+        val bitmapBuffer = Bitmap.createBitmap(
+            imageProxy.width, imageProxy.height, Bitmap.Config.ARGB_8888
+        )
+
+        imageProxy.use {
+            bitmapBuffer.copyPixelsFromBuffer(imageProxy.planes[0].buffer)
+        }
+        imageProxy.close()
+
+        // Used for rotating the frame image so it matches our models
+        val matrix = Matrix().apply {
+            postRotate(imageProxy.imageInfo.rotationDegrees.toFloat())
+        }
+
+        val rotatedBitmap = Bitmap.createBitmap(
+            bitmapBuffer,
+            0,
+            0,
+            bitmapBuffer.width,
+            bitmapBuffer.height,
+            matrix,
+            true
+        )
+
+        val mpImage = BitmapImageBuilder(rotatedBitmap).build()
+
+        segmentsAsync(mpImage, frameTime)
+    }
+
+    @VisibleForTesting
+    fun segmentsAsync(mpImage: MPImage, frameTime: Long) {
+        imagesegmenter?.segmentAsync(mpImage, frameTime)
+    }
+
+    // Runs image segmentation on single image and
+    // returns the results asynchronously to the caller.
```

Comment:
If this is returning asyncronously, can we add a comment above about the listeners being needed for all modes since it's different than classification/detection/recognition where the callbacks are only necessary for live streams